### PR TITLE
fix(streamlit): synchronize Advanced fields with YAML

### PR DESCRIPTION
**Pull request by liq22**
## User problem

Advanced mode rendered common-field widgets from the original template even after the user edited the full YAML draft. Because Streamlit keeps widget state by key, an older common-field value could silently override a newer YAML value during validation.

## Invariant

In Advanced mode, common-field controls are initialized from the current YAML draft. A YAML or template change refreshes the UI widget keys before validation; stale widget state cannot overwrite the new draft.

## Minimal change

- Parse the current Advanced YAML before rendering common fields.
- Use a monotonically increasing UI-only widget revision when the YAML or template changes.
- If YAML is invalid, withhold common-field editing until it is corrected.
- Keep Quick Start behavior unchanged.
- Add focused state tests and a real Streamlit AppTest covering YAML → common field → validation → YAML replacement.

The revision is not an experiment identity, digest, manifest, or backend authority.

## Out of scope

No backend resolver, Factory, Pipeline, runtime, split, objective, metric, checkpoint, THU/MFPT, batch scheduler, Agent execution, release/tag/package, or research PR change.

## Acceptance

Streamlit quality gates, Core quality gates, Repository layout, and Submodule policy must pass on the final head. AppTest must demonstrate that a newer YAML value replaces an older common-field widget state. Rollback is reverting this bounded PR.

### CODE REVIEW

File: apps/streamlit/ui_theme.py [hunk 1]
```diff
@@ -214,18 +251,29 @@ def _render_fields(
...
+            return ()
     specs = tuple(spec for spec in catalog.fields if spec.quick_start or not quick_only)
     values: Dict[str, Any] = {}
     columns = st.columns(min(4, max(1, len(specs))))
     for index, spec in enumerate(specs):
         with columns[index % len(columns)]:
             values[spec.key] = _render_field(
                 spec,
-                field_value(resolved, spec),
+                field_value(source, spec),
```
[the remark this answers is not in the document]
Comment (liq22) [reply apps/streamlit/ui_theme.py]: Addressed on the current head. Advanced common fields now pre-validate widget-facing types before rendering: integer controls reject floats/strings/bools, checkbox controls require bool, and text controls require text. Invalid YAML values are therefore preserved for the public inspector to reject instead of being repaired into an override. Added focused coercion tests plus AppTest coverage for `trainer.num_epochs: 1.5`, which must remain invalid and keep Run disabled.
